src/nio/client/async_client.py

```diff
@@ -3966,7 +3972,10 @@ async def room_upgrade(
                 ``m.room.power_levels`` event before it is sent to the room.
         """
         # Check if we are allowed to tombstone a room
-        if not await self.has_event_permission(old_room_id, "m.room.tombstone"):
+        tombstone_permission = await self.has_event_permission(
+            old_room_id, "m.room.tombstone", "state"
+        )
+        if tombstone_permission is not True:
```

Comment:
```suggestion
        if await self.has_event_permission(
            old_room_id, "m.room.tombstone", "state"
        ):
```

this seems neater

Reply:
Ah, right. I think it would be better to fail if `has_event_permission` returns an error, because in between the check and sending the tombstone event a new room is created, which imo is more annoying (for the user) to clean up than the tombstoning failing.

The TOC/TOU race condition seems fine to me, it's unlikely and is only mildly inconvenient if it happens. I suppose we could leave and forget the new room if the tombstone event fails to send, but that makes me nervous about somehow ending up in an unrecoverable scenario where the tombstone actually did succeed, but the user has now left the new room and can't rejoin...